### PR TITLE
libanki: more updates

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Purpose / Description

On top of #15476 

## Approach
In the commits

## How Has This Been Tested?

I'm gonna trust CI

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt [hunk 1]
```diff
@@ -99,7 +110,12 @@ interface TestClass {
 
     /** Adds a note with Text to Speech functionality */
     fun addNoteUsingTextToSpeechNoteType(front: String, back: String) {
-        addNonClozeModel("TTS", arrayOf("Front", "Back"), "{{Front}}{{tts en_GB:Front}}", "{{tts en_GB:Front}}<br>{{Back}}")
+        addNonClozeModel(
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/testutils/TestClass.kt:113]: Spacing-only changes

File: AnkiDroid/src/main/java/com/ichi2/libanki/Note.kt [hunk 2]
```diff
@@ -88,8 +89,9 @@ class Note : Cloneable {
         this.fMap = Notetypes.fieldMap(notetype)
     }
 
+    @NotInLibAnki
     fun numberOfCards(col: Collection): Int {
-        return col.db.queryLongScalar("SELECT count() FROM cards WHERE nid = ?", this.id).toInt()
+        return cardIds(col).size
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Note.kt:96]: could you check the performance of this one?

File: AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt [hunk 3]
```diff
@@ -666,10 +666,6 @@ open class Collection(
         return backend.updateNotes(notes = notes.map { it.toBackendNote() }, skipUndoEntry = false)
     }
 
-    fun emptyCids(): List<CardId> {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Collection.kt:669]: can we move this to a non-libAnki extension function, it seems useful still

**Review by david-allison — APPROVED** [review]
Cheers!

**Comment by lukstbit** [comment]
Took a conflict from the linked PR.

**Review by lukstbit — APPROVED** [review]

**Comment by lukstbit** [comment]
Ok, also reviewed it. Feel free to merge after fixing the conflicts.

**Pull request merged**